nxontology_ml/features.py

```diff
@@ -0,0 +1,40 @@
+from collections.abc import Iterator
+
+import pandas as pd
+from nxontology import NXOntology
+from nxontology.node import Node as T_Node
+from nxontology.node import Node_Info
+
+
+class NodeFeatures:
+    def __init__(self, info: Node_Info[T_Node]) -> None:
+        self.info = info
```

Comment:
Instead of a separate `NodeFeatures` class that embeds a node info object, I think a subclass of Node_Info could be nice. This would make it really easy for us to promote methods from `nxontology_ml` to `nxontology`, and it would allow us to benefit from the caching infrastructure provided by nxontology.

Basic approach in https://github.com/related-sciences/nxontology/pull/26, but timeline on getting that working with typing and reviewing its desirability is TBD.